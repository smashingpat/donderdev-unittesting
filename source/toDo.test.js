import createTodo from './todo.js';


jest.mock('./random.js', () => {
    let count = 0;
    return () => {
        count += 0.001;
        return count;
    }
})

describe('createTodo', () => {
    it('should create a todo', () => {
        const todoItem = createTodo();

        expect(todoItem).toMatchSnapshot();
    });

    it('should create a todo with text as the argument', () => {
        const result = createTodo('write a test')
        expect(result.text).toBe('write a test');
    });

    it('should create a random id', () => {
        const result = createTodo('write a test');
        const result2 = createTodo('write another test');

        expect(result.id).not.toBe(result2.id);
    });
});