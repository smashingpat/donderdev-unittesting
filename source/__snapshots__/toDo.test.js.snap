// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTodo should create a todo 1`] = `
Object {
  "id": 10000000,
  "status": true,
  "text": "",
}
`;
